core/chains/evm/gas/models.go

```diff
@@ -36,11 +36,11 @@ type EvmFeeEstimator interface {
 
 	// L1Oracle returns the L1 gas price oracle only if the chain has one, e.g. OP stack L2s and Arbitrum.
 	L1Oracle() rollups.L1Oracle
-	GetFee(ctx context.Context, calldata []byte, feeLimit uint64, maxFeePrice *assets.Wei, toAddress *common.Address, opts ...feetypes.Opt) (fee EvmFee, estimatedFeeLimit uint64, err error)
+	GetFee(ctx context.Context, calldata []byte, feeLimit uint64, maxFeePrice *assets.Wei, fromAddress, toAddress *common.Address, opts ...feetypes.Opt) (fee EvmFee, estimatedFeeLimit uint64, err error)
```

Comment:
I'm not a big fan of adding more and more fields in the signature. At some point, it might make sense to just pass the transaction and access the fields internally, but I guess it's ok for now.

Reply:
Ya I agree. I even considered splitting gas price estimation from gas limit estimation. Externally, it seems for the most part products and CW only care about the gas price and have a separate method for limits so we could remove `calldata`, `feeLimit`, `fromAddress`, and `toAddress`. But think we're bound by Arbitrum's estimator for now so maybe it's something we can cleanup in the future